### PR TITLE
107895 avoid sin 4 getting all columns - Vol 2

### PR DESCRIPTION
**Pull request by janid1967**
[AB#107895](https://statoildeveloper.visualstudio.com/b6d97460-2e53-47ae-aad4-b41396d90828/_workitems/edit/107895)

As already impl for PunchItemVallidator in 107501

### CODE REVIEW

File: src/Equinor.ProCoSys.Completion.Command/Validators/DocumentValidator.cs [hunk 1]
```diff
@@ -13,19 +13,18 @@ public class DocumentValidator : IDocumentValidator
...
-            where l.Guid == documentGuid
-            select l).AnyAsync(cancellationToken);
-
-    public async Task<bool> IsVoidedAsync(Guid documentGuid, CancellationToken cancellationToken)
-    {
-        var document = await(from d in _context.QuerySet<Document>()
+        await (from d in _context.QuerySet<Document>()
             where d.Guid == documentGuid
-            select d).SingleOrDefaultAsync(cancellationToken);
+            select 1).AnyAsync(cancellationToken);
```
Comment (jsoi-eq) [inline src/Equinor.ProCoSys.Completion.Command/Validators/DocumentValidator.cs:24]: We lose the "fail fast" we get from SingleOrDefault when there are multiple results when we only expect 1 result.
Has this tradeoff been evaluated and concluded that the optimization is more worth than the "fail fast".

(I have been burnt more times than I care to count from a faulty query filter leading to multiple results and picking an incorrect result value)

**Review by jsoi-eq — COMMENTED** [review]
Awaiting feedback.

File: src/Equinor.ProCoSys.Completion.Command/Validators/DocumentValidator.cs (hunk 1 above)
Comment (janid1967) [reply src/Equinor.ProCoSys.Completion.Command/Validators/DocumentValidator.cs:24]: This is more about how EF create the SQL. 
Using AnyAsync, EF produces DB queries as 
```
SELECT
    CASE WHEN EXISTS (
        SELECT 1
        FROM YourTableName AS t
        WHERE t.SomeColumn = @someValue
    ) THEN CAST(1 AS bit)
    ELSE CAST(0 AS bit) END
```
SingleOrDefaulltAsync produces DB query with top(2) of columns selected

EXISTS is recommended to check if item exists in DB.

I've measured 100 selects in a row for each method. Any seem to be slightly faster in my box

**Comment by janid1967** [comment]
After team discussion we agree not to do this optimization

**Pull request closed without merging**